src/Streamly/Internal/Data/Stream/IsStream.hs

```diff
@@ -4602,35 +4604,33 @@ data SessionState t m k a b = SessionState
 --
 -- /Internal/
 --
-
--- XXX Perhaps we should use an "Event a" type to represent timestamped data.
 {-# INLINABLE classifySessionsBy #-}
-classifySessionsBy
-    :: (IsStream t, MonadAsync m, Ord k)
-    => Double         -- ^ timer tick in seconds
-    -> Double         -- ^ session timeout in seconds
-    -> Bool           -- ^ reset the timeout when an event is received
+classifySessionsBy ::
+       (IsStream t, MonadAsync m, Ord k)
+    => Double -- ^ timer tick in seconds
+    -> Double -- ^ session timeout in seconds
+    -> Bool -- ^ reset the timeout when an event is received
     -> (Int -> m Bool) -- ^ predicate to eject sessions based on session count
-    -> Fold m a (Either b b) -- ^ Fold to be applied to session events
+    -> Fold m a b -- ^ Fold to be applied to session events
     -> t m (k, a, AbsTime) -- ^ session key, data, timestamp
     -> t m (k, b) -- ^ session key, fold result
-classifySessionsBy tick tmout reset ejectPred
-    (Fold step initial extract) str =
-    concatMap sessionOutputStream $
-        scanlMAfter' sstep (return szero) flush stream
+classifySessionsBy tick tmout reset ejectPred (Fold step initial extract) str =
+    concatMap sessionOutputStream
+      $ scanlMAfter' sstep (return szero) flush stream
 
     where
 
     timeoutMs = toRelTime (round (tmout * 1000) :: MilliSecond64)
     tickMs = toRelTime (round (tick * 1000) :: MilliSecond64)
-    szero = SessionState
-        { sessionCurTime = toAbsTime (0 :: MilliSecond64)
-        , sessionEventTime = toAbsTime (0 :: MilliSecond64)
-        , sessionCount = 0
-        , sessionTimerHeap = H.empty
-        , sessionKeyValueMap = Map.empty
-        , sessionOutputStream = K.nil
-        }
+    szero =
```

Comment:
I believe I messed up the formatting of `Record`. I'll change it in the next commit.